pipeline/configuration.py

```diff
@@ -0,0 +1,60 @@
+import base64
+import json
+import os
+import sys
+from pathlib import Path
+from platform import python_version
+from typing import TypedDict
+
+from packaging import version
+
+from pipeline.util.logging import _print
+
+PIPELINE_CACHE = os.getenv(
+    "PIPELINE_CACHE",
+    os.path.join(os.getenv("APPDATA"), "/Local/.pipeline/")
+    if (sys.platform == "win32" or sys.platform == "cygwin")
+    else os.path.join(str(Path.home()), ".cache/pipeline/"),
+)
+
+if not os.path.exists(PIPELINE_CACHE):
```

Comment:
I'm not really a fan of a library creating files on my system unexpectedly. As far as I'm aware, this will create a new directory whenever this module is imported, even if not using the CLI. I think it would be better to only create the directory if using the CLI for auth.

Reply:
👍